stdlib/public/core/HashTable.swift

```diff
@@ -82,6 +82,32 @@ extension _HashTable {
     // The scale is the exponent corresponding to the bucket count.
     return exponent
   }
+
+  internal static func hashSeed(
+    for object: AnyObject,
+    scale: Int
+  ) -> Int {
+    // We generate a new hash seed whenever a new hash table is allocated and
+    // whenever an existing table is resized, so that we avoid certain copy
+    // operations becoming quadratic.  (For background details, see
+    // https://bugs.swift.org/browse/SR-3268)
+    //
+    // Note that we do reuse the existing seed when making copy-on-write copies
+    // so that we avoid breaking value semantics.
+    if Hasher._isDeterministic {
+      // When we're using deterministic hashing, the scale value as the seed is
+      // still allowed, and it covers most cases. (Unfortunately some operations
+      // that merge two similar-sized hash tables will still be quadratic.)
+      return scale
```

Comment:
If we wanted to get rid of quadratic behavior in the deterministic hashing case, we would use an atomic counter here, I guess? Do we care?

Reply:
That's a really clever idea. Not sure it's worth the trouble though -- deterministic hashing is intended as a debugging/testing tool, and it's supposed to never be enabled in production.

What if we used an atomic counter for *all* cases?